plugins/database/mongodb/connection_producer.go

```diff
@@ -73,18 +79,41 @@ func (c *mongoDBConnectionProducer) Init(ctx context.Context, conf map[string]in
 			input = string(inputBytes)
 		}
 
-		concern := &mgo.Safe{}
+		concern := &writeConcern{}
 		err = json.Unmarshal([]byte(input), concern)
 		if err != nil {
-			return nil, errwrap.Wrapf("error mashalling write_concern: {{err}}", err)
+			return nil, errwrap.Wrapf("error marshalling write_concern: {{err}}", err)
+		}
+
+		// Translate write concern to mongo options
+		var w func(*writeconcern.WriteConcern)
```

Comment:
Nit: Can `w` and `j` be defined as type `writeconcern.Option` instead of `func(*writeconcern.WriteConcern)`?

Reply:
Thanks!